sentry-core/src/main/java/io/sentry/core/protocol/Device.java

```diff
@@ -343,9 +346,35 @@ public void setBatteryTemperature(Float batteryTemperature) {
     LANDSCAPE
   }
 
+  @TestOnly
+  Map<String, Object> getUnknown() {
+    return unknown;
+  }
+
   @ApiStatus.Internal
   @Override
   public void acceptUnknownProperties(Map<String, Object> unknown) {
     this.unknown = unknown;
   }
+
+  /**
+   * Clones a Device aka deep copy
+   *
+   * @return the cloned Device
+   * @throws CloneNotSupportedException if object is not cloneable
+   */
+  @Override
+  public @NotNull Device clone() throws CloneNotSupportedException {
+    final Device clone = (Device) super.clone();
+
+    final String[] archsRef = this.archs;
+    clone.archs = archsRef != null ? this.archs.clone() : null;
+
+    final TimeZone timezoneRef = this.timezone;
+    clone.timezone = timezoneRef != null ? (TimeZone) this.timezone.clone() : null;
+
```

Comment:
I think you need to deep copy `orientation` here too, like:

```
    final SentryLevel levelRef = level;
    clone.level =
        levelRef != null ? SentryLevel.valueOf(levelRef.name().toUpperCase(Locale.ROOT)) : null;
```

Reply:
Why is this part `.toUpperCase(Locale.ROOT)` needed?

Reply:
Enum is immutable so that's fine to use the same ref

Reply:
I remember having tests failing because it was the same ref. and recreating it using `valueOf` fixed the issue.
we had a short time to do this deep copy methods so it might be an oversight too.